Packs/GenericSQL/Integrations/GenericSQL/GenericSQL.py

```diff
@@ -178,27 +189,62 @@ def generate_default_port_by_dialect(dialect: str) -> Optional[str]:
     return None
 
 
-def generate_bind_vars(bind_variables_names: str, bind_variables_values: str) -> Any:
+def generate_variable_names_and_mapping(bind_variables_values_list: list, query: str, dialect: str) ->\
+        tuple[dict[str, Any], str | Any]:
+    """
+    In case of passing just bind_variables_values, since it's no longer supported in SQL Alchemy v2.,
+    this function generates names for those variables and return an edited query with a mapping.
+    Args:
+        bind_variables_values_list: Values to put in the bind variables
+        query: The given query which contains chars to replace
+        dialect: The DB dialect
+
+    Returns: A mapping (dict) and an edited query.
+
+    """
+    mapping_dialect_regex = {MICROSOFT_SQL_SERVER: "\\?",
+                             MS_ODBC_DRIVER: "\\?",
+                             POSTGRES_SQL: "%s",
+                             MY_SQL: "%s"
+                             }
+    try:
+        char_to_replace = mapping_dialect_regex[dialect]
+    except Exception as e:
+        raise DemistoException(
+            "Placeholder was not found for the given dialect"
+        ) from e
```

Comment:
instead of try/except, check `if dialect in mapping_dialect_regex: ... else: raise ...`